sycl/include/CL/sycl/device.hpp

```diff
@@ -176,7 +176,7 @@ class __SYCL_EXPORT device {
   /// \return a native handle, the type of which defined by the backend.
   template <backend BackendName>
   auto get_native() const -> typename interop<BackendName, device>::type {
-    return static_cast<typename interop<BackendName, device>::type>(
+    return reinterpret_cast<typename interop<BackendName, device>::type>(
```

Comment:
Interesting... `cl_device_id` -> `pi_device` cast seems to be forbidden. But the opposite direction should be fine. Can we use `pi::cast` here?

Reply:
this is converting from pi_native_handle to native handle, the pi::cast should work and be appropriate.